panic: runtime error: slice bounds out of range upon connection from Pod

**Describe the bug**
When replacing a Deployment with telepresence the local root daemon crashes upon the first connection received.

**To Reproduce**
I run the following sequence of commands:
```
> telepresence connect -n {{target-namespace}}
Connected to context arn:aws:eks:eu-central-1:...
```

```
> telepresence replace {{deployment-name}} --env-file example-app.env
Using Deployment {{deployment-name}}
   Container name    : {{container-name}}
   State             : ACTIVE
   Workload kind     : Deployment
   Port forwards     : 2a05:d014:73c:7c02:a6d0::15 -> 127.0.0.1
       50051 -> 50051 TCP
   Volume Mount Point: /var/folders/g0/bjy50r2j19j1w1vwpl7ckqdw0000gn/T/telfs-3296985765
```

```
> telepresence status
OSS User Daemon: Running
  Version           : 2.22.1
  Executable        : /opt/homebrew/bin/telepresence
  Install ID        : 55877e3b-5045-4a40-b74e-f1826fddcf47
  Status            : Connected
  Kubernetes server : https://{{cluster}}.yl4.eks-cluster.eu-central-1.api.aws
  Kubernetes context: arn:aws:eks:...
  Namespace         : {{namespace}}
  Manager namespace : ambassador
  Mapped namespaces : [...]
  Intercepts        : 1 total
    ingest/ingest: ...
OSS Root Daemon: Running
  Version: v2.22.1
  DNS    : 
    Local IP        : 127.0.0.1
    Exclude suffixes: [.com .io .net .org .ru]
    Include suffixes: []
    Timeout         : 4s
  Subnets: (2 subnets)
    - fd9f:f2fe:af9c::/108
    - 2a05:d014:73c:7c02:a6d0::/120
OSS Traffic Manager: Connected
  Version      : v2.22.1
  Traffic Agent: ghcr.io/telepresenceio/tel2:2.22.1
```

Connection is up, DNS works, I can access cluster internal services.

Next I send a request to the replaced Deployment and see the following in daemon.log
```
panic: runtime error: slice bounds out of range [:34] with length 25

goroutine 827 [running]:
github.com/telepresenceio/telepresence/v2/pkg/tunnel.ConnID.IsDestinationIPv4(...)
        github.com/telepresenceio/telepresence/v2/pkg/tunnel/connid.go:71
github.com/telepresenceio/telepresence/v2/pkg/tunnel.ConnID.DestinationProtocolString({0x14000ba0160?, 0x14001632280?})
        github.com/telepresenceio/telepresence/v2/pkg/tunnel/connid.go:150 +0x1a4
github.com/telepresenceio/telepresence/v2/pkg/tunnel.(*dialer).Start.func1()
        github.com/telepresenceio/telepresence/v2/pkg/tunnel/dialer.go:139 +0x25c
created by github.com/telepresenceio/telepresence/v2/pkg/tunnel.(*dialer).Start in goroutine 825
        github.com/telepresenceio/telepresence/v2/pkg/tunnel/dialer.go:126 +0x7c
```

I retrieve an HTTP 503 (I assume from the ingest controller since my local service never gets hit). Status subsequently shows the local daemon as not running anymore.

```
> telepresence status
OSS User Daemon: Running
  Version           : 2.22.1
  Executable        : /opt/homebrew/bin/telepresence
  Install ID        : 55877e3b-5045-4a40-b74e-f1826fddcf47
  Status            : 
  Kubernetes server : 
  Kubernetes context: 
  Namespace         : 
  Manager namespace : 
Root Daemon: Not running
Traffic Manager: Not connected
```

**Expected behavior**
My local service receiving traffic from within the cluster.

**Versions:**
```
> telepresence version
OSS Client         : v2.22.1
OSS Root Daemon    : v2.22.1
OSS User Daemon    : v2.22.1
OSS Traffic Manager: v2.22.1
Traffic Agent      : ghcr.io/telepresenceio/tel2:2.22.1
```

Up-to-date MBP M2, targeted Kubernetes is AWS EKS cluster (v1.32, eks.5).

**Additional context**
Cluster is IPV6-only on arm64 nodes (t4g)

--- a/go.mod
+++ b/go.mod
@@ -39,7 +39,7 @@ require (
 	github.com/stretchr/testify v1.10.0
 	github.com/telepresenceio/go-fuseftp v0.6.6
 	github.com/telepresenceio/go-fuseftp/rpc v0.6.6
-	github.com/telepresenceio/telepresence/rpc/v2 v2.22.1
+	github.com/telepresenceio/telepresence/rpc/v2 v2.22.2-test.0
 	github.com/vishvananda/netlink v1.3.0
 	golang.org/x/net v0.37.0
 	golang.org/x/sys v0.31.0

--- a/pkg/tunnel/connid.go
+++ b/pkg/tunnel/connid.go
@@ -21,15 +21,24 @@ func ConnIDFromUDP(src, dst *net.UDPAddr) ConnID {
 func NewConnID(proto int, src, dst netip.AddrPort) ConnID {
 	srcAddr := src.Addr()
 	dstAddr := dst.Addr()
-	if srcAddr.Is4In6() {
+	switch {
+	case srcAddr.Is4():
+		if dstAddr.Is4In6() {
+			dstAddr = dstAddr.Unmap()
+		} else if dstAddr.Is6() {
+			srcAddr = netip.AddrFrom16(srcAddr.As16())
+		}
+	case srcAddr.Is4In6():
 		if dstAddr.Is4() {
 			srcAddr = srcAddr.Unmap()
 		} else if dstAddr.Is4In6() {
 			srcAddr = srcAddr.Unmap()
 			dstAddr = dstAddr.Unmap()
 		}
-	} else if srcAddr.Is4() && dstAddr.Is4In6() {
-		dstAddr = dstAddr.Unmap()
+	default:
+		if dstAddr.Is4() {
+			dstAddr = netip.AddrFrom16(dstAddr.As16())
+		}
 	}
 
 	ls := srcAddr.BitLen() / 8
@@ -63,12 +72,12 @@ func (id ConnID) areBothIPv4() bool {
 
 // IsSourceIPv4 returns true if the source of this ConnID is IPv4.
 func (id ConnID) IsSourceIPv4() bool {
-	return id.areBothIPv4() || net.IP(id[0:16]).To4() != nil
+	return id.areBothIPv4() || len(id) > 16 && net.IP(id[0:16]).To4() != nil
 }
 
 // IsDestinationIPv4 returns true if the destination of this ConnID is IPv4.
 func (id ConnID) IsDestinationIPv4() bool {
-	return id.areBothIPv4() || net.IP(id[18:34]).To4() != nil
+	return id.areBothIPv4() || len(id) == 37 && net.IP(id[18:34]).To4() != nil
 }
 
 // Source returns the source address and port.

--- a/pkg/tunnel/connid_test.go
+++ b/pkg/tunnel/connid_test.go
@@ -1,0 +1,296 @@
+package tunnel
+
+import (
+	"net"
+	"net/netip"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/telepresenceio/telepresence/v2/pkg/ipproto"
+)
+
+var (
+	ipv4a     = netip.MustParseAddr("192.168.0.1")
+	ipv4b     = netip.MustParseAddr("192.168.3.8")
+	ipv6a     = netip.MustParseAddr("2a05:d014:153d:d732:a2d3::15")
+	ipv6b     = netip.MustParseAddr("2a05:d014:153d:d732:a2f2::08")
+	ipv4Asv6a = netip.AddrFrom16(ipv4a.As16())
+	ipv4Asv6b = netip.AddrFrom16(ipv4b.As16())
+)
+
+func TestConnIDFromUDP(t *testing.T) {
+	tests := []struct {
+		name string
+		src  *net.UDPAddr
+		dst  *net.UDPAddr
+		want string
+	}{
+		{
+			name: "ipv4-ipv4",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> 192.168.3.8:8",
+		},
+		{
+			name: "ipv4-ipv6",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> [2a05:d014:153d:d732:a2f2::8]:8",
+		},
+		{
+			name: "ipv4-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> 192.168.3.8:8",
+		},
+		{
+			name: "ipv6-ipv4",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: "udp [2a05:d014:153d:d732:a2d3::15]:4 -> 192.168.3.8:8",
+		},
+		{
+			name: "ipv6-ipv6",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: "udp [2a05:d014:153d:d732:a2d3::15]:4 -> [2a05:d014:153d:d732:a2f2::8]:8",
+		},
+		{
+			name: "ipv6-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: "udp [2a05:d014:153d:d732:a2d3::15]:4 -> 192.168.3.8:8",
+		},
+		{
+			name: "ipv4as6-ipv4",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> 192.168.3.8:8",
+		},
+		{
+			name: "ipv4as6-ipv6",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> [2a05:d014:153d:d732:a2f2::8]:8",
+		},
+		{
+			name: "ipv4as6-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: "udp 192.168.0.1:4 -> 192.168.3.8:8",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, ConnIDFromUDP(tt.src, tt.dst).String(), "ConnIDFromUDP(%v, %v)", tt.src, tt.dst)
+		})
+	}
+}
+
+func TestConnID_Source(t *testing.T) {
+	tests := []struct {
+		name string
+		src  netip.AddrPort
+		dst  netip.AddrPort
+		want netip.AddrPort
+	}{
+		{
+			name: "ipv4-ipv4",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+		{
+			name: "ipv4-ipv6",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+		{
+			name: "ipv4-ipv4as6",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+		{
+			name: "ipv6-ipv4",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv6a, 4),
+		},
+		{
+			name: "ipv6-ipv6",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv6a, 4),
+		},
+		{
+			name: "ipv6-ipv4as6",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv6a, 4),
+		},
+		{
+			name: "ipv4as6-ipv4",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+		{
+			name: "ipv4as6-ipv6",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+		{
+			name: "ipv4as6-ipv4as6",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv4a, 4),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, NewConnID(ipproto.TCP, tt.src, tt.dst).Source(), "Source()")
+		})
+	}
+}
+
+func TestConnID_Destination(t *testing.T) {
+	tests := []struct {
+		name string
+		src  netip.AddrPort
+		dst  netip.AddrPort
+		want netip.AddrPort
+	}{
+		{
+			name: "ipv4-ipv4",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+		{
+			name: "ipv4-ipv6",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv6b, 8),
+		},
+		{
+			name: "ipv4-ipv4as6",
+			src:  netip.AddrPortFrom(ipv4a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+		{
+			name: "ipv6-ipv4",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+		{
+			name: "ipv6-ipv6",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv6b, 8),
+		},
+		{
+			name: "ipv6-ipv4as6",
+			src:  netip.AddrPortFrom(ipv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+		{
+			name: "ipv4as6-ipv4",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+		{
+			name: "ipv4as6-ipv6",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv6b, 8),
+			want: netip.AddrPortFrom(ipv6b, 8),
+		},
+		{
+			name: "ipv4as6-ipv4as6",
+			src:  netip.AddrPortFrom(ipv4Asv6a, 4),
+			dst:  netip.AddrPortFrom(ipv4Asv6b, 8),
+			want: netip.AddrPortFrom(ipv4b, 8),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, NewConnID(ipproto.TCP, tt.src, tt.dst).Destination(), "Source()")
+		})
+	}
+}
+
+func TestConnID_areBothIPv4(t *testing.T) {
+	tests := []struct {
+		name string
+		src  *net.UDPAddr
+		dst  *net.UDPAddr
+		want bool
+	}{
+		{
+			name: "ipv4-ipv4",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: true,
+		},
+		{
+			name: "ipv4-ipv6",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: false,
+		},
+		{
+			name: "ipv4-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv4a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: true,
+		},
+		{
+			name: "ipv6-ipv4",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: false,
+		},
+		{
+			name: "ipv6-ipv6",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: false,
+		},
+		{
+			name: "ipv6-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: false,
+		},
+		{
+			name: "ipv4as6-ipv4",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4b.AsSlice(), Port: 8},
+			want: true,
+		},
+		{
+			name: "ipv4as6-ipv6",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv6b.AsSlice(), Port: 8},
+			want: false,
+		},
+		{
+			name: "ipv4as6-ipv4as6",
+			src:  &net.UDPAddr{IP: ipv4Asv6a.AsSlice(), Port: 4},
+			dst:  &net.UDPAddr{IP: ipv4Asv6b.AsSlice(), Port: 8},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, ConnIDFromUDP(tt.src, tt.dst).areBothIPv4(), "ConnIDFromUDP(%v, %v)", tt.src, tt.dst)
+		})
+	}
+}

--- a/pkg/vif/testdata/router/go.mod
+++ b/pkg/vif/testdata/router/go.mod
@@ -51,7 +51,7 @@ require (
 	github.com/rogpeppe/go-internal v1.14.1 // indirect
 	github.com/spf13/cobra v1.9.1 // indirect
 	github.com/spf13/pflag v1.0.6 // indirect
-	github.com/telepresenceio/telepresence/rpc/v2 v2.22.1 // indirect
+	github.com/telepresenceio/telepresence/rpc/v2 v2.22.2-test.0 // indirect
 	github.com/vishvananda/netlink v1.3.0 // indirect
 	github.com/vishvananda/netns v0.0.5 // indirect
 	github.com/x448/float16 v0.8.4 // indirect

